### PR TITLE
plan: remove `DatsSource.statsAfterSelect`.

### PR DESCRIPTION
**Pull request by winoros**
## What have you changed? (mandatory)

Currently, `DataSource` have two stats, one is derived from `basePlan`, one is `statsAfterSelect`.
These two actually can be merged into one. And so we can remove `statsAfterSelect`.

## What is the type of the changes? (mandatory)

- tiny refactor

## How has this PR been tested? (mandatory)

existing tests.

## Refer to a related PR or issue link (optional)

#7049

### CODE REVIEW

File: plan/stats.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func (s *statsInfo) scale(factor float64) *statsInfo {
 }
 
 // We try to scale statsInfo to an expectCnt which must be smaller than the derived cnt.
+// TODO: try to use a better way to do this.
```
Comment (zz-jason) [inline plan/stats.go]: need to add a issue to this TODO and describe it in detail.

Comment (winoros) [reply plan/stats.go]: This should be in proposal of cost model(or another doc that introduce the `statsInfo`). I think don't need a another issue.

File: plan/find_best_task.go [hunk 2]
```diff
@@ -394,13 +395,14 @@ func (ds *DataSource) convertToIndexScan(prop *requiredProp, path *accessPath) (
...
 	// If expectedCnt and count2 are both zero and we go into the below `if` block, the count1 will be set to zero though it's shouldn't be.
-	if (matchProperty || prop.isEmpty()) && prop.expectedCnt < ds.statsAfterSelect.count {
-		selectivity := ds.statsAfterSelect.count / path.countAfterAccess
+	if (matchProperty || prop.isEmpty()) && prop.expectedCnt < ds.stats.count {
+		selectivity := ds.stats.count / path.countAfterAccess
 		rowCount = math.Min(prop.expectedCnt/selectivity, rowCount)
 	}
-	is.stats = ds.stats.scaleByExpectCnt(rowCount)
+	is.stats = newSimpleStats(rowCount)
+	logrus.Warnf("coutn: %v", is.stats.count)
```
Comment (lamxTyler) [inline plan/find_best_task.go]: Remove the debug log.

**Comment by winoros** [comment]
/run-all-tests
PTAL @zz-jason

**Pull request merged**